backend/src/services/lectures/lectures.repo.js

```diff
@@ -101,9 +151,20 @@ class LecturesRepo {
           : new Date(data.lecture_date);
     }
 
-    // Add code if provided
-    if (data.code !== undefined) {
+    // Handle code regeneration
+    if (data.regenerate_code === true && data.course_id) {
+      const newCode = await this.generateUniqueCode(data.course_id);
+      updateData.code = newCode;
+      // Set expiration timestamps: code valid for 5 minutes
+      updateData.code_generated_at = new Date();
+      updateData.code_expires_at = new Date(
+        updateData.code_generated_at.getTime() + 5 * 60 * 1000
+      ); // 5 minutes
+    } else if (data.code !== undefined) {
```

Comment:
[Suggestion]
This is doing the exact same update as the activateAttendance endpoint. I think in this case, it would be better then to make this explicit. Let's remove the ability to regenerate code (since there is no code to begin with or if a code exists and attendance has been taken already, we need to decide if the lecture can take attendance again) and have the client explicitly call activateAttendance to get a code.